design/cluster-scope-resource-filter.md

```diff
@@ -0,0 +1,572 @@
+
+
+## High-Level Design
+Four new parameters are added into command `velero backup create`: `--include-cluster-scope-resources`, `--exclude-cluster-scope-resources`, `--include-namespaced-resources` and `--exclude-namespaced-resources`. 
+`--include-cluster-scope-resources` and `--exclude-cluster-scope-resources` are used to filter cluster scope resources included or excluded in backup per resource type.
+`--include-namespaced-resources` and `--exclude-namespaced-resources` are used to filter namespace scope resources included or excluded in backup per resource type.
+
+### Parameters Rules
+
+* `--include-cluster-scope-resources`, `--include-namespaced-resources` and `--exclude-namespaced-resources` valid value include `*` and comma separated string.
+
+* `--exclude-namespaces` also supports `*`.
+
+* `--exclude-cluster-scope-resources` valid value is comma separated string.
+
+* `--include-cluster-scope-resources` and `--exclude-cluster-scope-resources` can only contain cluster scope resource type names. If namespace scope resource type names are included, validation failure should be returned.
+
+* If there are conflicts between `--include-cluster-scope-resources` and `--exclude-cluster-scope-resources` specified resources type lists, validation failure should be returned.
+
+* `--include-namespaced-resources` and `--exclude-namespaced-resources` can only contain namespace scope resource type names. If cluster scope resource type names are included, validation failure should be returned.
+
+* If there are conflicts between `--include-namespaced-resources` and `--exclude-namespaced-resources` specified resources type lists, validation failure should be returned.
+
+* If  `--include-namespaced-resources` is not present, it means including all namespace scope resources per resource type.
+
+* If `--include-cluster-scope-resources` is not present, it means no additional cluster resource is included per resource type.
+
+* `--include-cluster-scope-resources`, `--exclude-cluster-scope-resources`, `--include-namespaced-resources` and `--exclude-namespaced-resources` have interact with `--include-resources` and `--exclude-resources`. The rule is specified in [IncludedResources and ExcludedResources change](#includedresources-and-excludedresources-change)
+
+* `--include-cluster-scope-resources` and `--exclude-cluster-scope-resources` have interact with `--include-cluster-resources` parameter. The rule is specified in [IncludeClusterResources change](#includeclusterresources-change)
+
+* If the backup or restore contains no resource, validation failure should be returned.
+
+### Using scenarios:
+#### no namespaced resources + no cluster resources
+This is not allowed. Backup or restore cannot contain no resource.
+
+#### no namespaced resources + some cluster resources
+The following commands all mean backup no namespaced resources and some cluster resources.
+
+``` bash
+velero backup create <backup-name>
+	--exclude-namespaces=*
```

Comment:
I don't think wildcards are currently supported for exclude namespaces. However, instead of doing this, it would be clearer to simply leave `--include-namespaced-resources` empty, if the behavior is modified according to my suggestion above.